src/main/js/tansaction/transaction-form.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Transaction form.
+ *
+ * @author Xiaobing Hou
+ */
+
+import React from "react";
+import { ErrorMessage, Field, Form, Formik } from "formik";
+import { Button, FormGroup, FormLabel } from "react-bootstrap";
+import * as Yup from "yup";
+
+const TransactionForm = (props) => {
+  return (
+    <Formik
+      initialValues={{
+        bankAccountId: props.bankAccountId,
+        money: props.isAddTransaction ? 0 : props.selectedTransaction.money,
+        counterparty: props.isAddTransaction
+          ? ""
+          : props.selectedTransaction.counterparty,
+        time: props.isAddTransaction ? "" : props.selectedTransaction.time,
+        transactionDescription: props.isAddTransaction
+          ? ""
+          : props.selectedTransaction.transactionDescription,
+      }}
+      validationSchema={Yup.object({
+        money: Yup.number().required("Required"),
+        counterparty: Yup.string()
+          .max(100, "Must be 100 characters or less")
+          .required("Required"),
+        time: Yup.string()
+          .max(40, "Must be 40 characters or less")
+          .required("Required"),
+        transactionDescription: Yup.string().max(
+          100,
+          "Must be 100 characters or less"
+        ),
+      })}
+      onSubmit={(values, actions) => {
+        if (props.isAddTransaction) {
```

Comment:
Logic like this is better off being kept in the container. Remember the form is just presentation and should have logic about how you are going to process it.